src/Workspaces/Core/Portable/Diagnostics/DiagnosticProvider.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.SolutionCrawler;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Provide a way for users to turn on and off analyzing workspace for compiler diagnostics
+    /// </summary>
+    internal static class DiagnosticProvider
+    {
+        public static void Enable(Workspace workspace, Options options)
```

Comment:
should this API at VS layer since its effect can only be observed from editor layer and up.